src/main/java/mate/academy/springboot/criteriaquery/controller/PhoneController.java

```diff
@@ -0,0 +1,97 @@
+package mate.academy.springboot.criteriaquery.controller;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+import java.util.Set;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import mate.academy.springboot.criteriaquery.dto.PhoneRequestDto;
+import mate.academy.springboot.criteriaquery.dto.PhoneResponseDto;
+import mate.academy.springboot.criteriaquery.dto.mapper.PhoneMapper;
+import mate.academy.springboot.criteriaquery.model.Feature;
+import mate.academy.springboot.criteriaquery.model.Maker;
+import mate.academy.springboot.criteriaquery.model.Phone;
+import mate.academy.springboot.criteriaquery.service.FeatureService;
+import mate.academy.springboot.criteriaquery.service.MakerService;
+import mate.academy.springboot.criteriaquery.service.PhoneService;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/phones")
+public class PhoneController {
+    private final PhoneMapper phoneMapper;
+    private final PhoneService phoneService;
+    private final MakerService makerService;
```

Comment:
```suggestion
    private final ManufacturerService makerService;
```
mareService трохи неприродньо звучить імхо